velox/functions/sparksql/String.h

```diff
@@ -1233,4 +1233,169 @@ struct SoundexFunction {
+   * 5 | | | | |#| --> lower boundary:4, upper:4;
+   * 6 | | | | | |
+   * </pre>
+   */
+  template <typename TCodePoint>
+  void doCall(
+      out_type<int32_t>& result,
+      const TCodePoint* leftCodePoints,
+      const TCodePoint* rightCodePoints,
+      size_t leftCodePointsSize,
+      size_t rightCodePointsSize,
+      int32_t threshold) {
+    if (leftCodePointsSize < rightCodePointsSize) {
+      doCall(
+          result,
+          rightCodePoints,
+          leftCodePoints,
+          rightCodePointsSize,
+          leftCodePointsSize,
+          threshold);
+      return;
+    }
+    VELOX_USER_CHECK_LE(
+        leftCodePointsSize,
+        INT32_MAX,
+        "The inputs size exceeded max Levenshtein distance input size,"
+        " the code points size of left is {}, code points size of right is {}",
+        leftCodePointsSize,
+        rightCodePointsSize);
+    if (leftCodePointsSize - rightCodePointsSize > threshold) {
+      result = -1;
+      return;
+    }
+    if (rightCodePointsSize == 0) {
+      result = leftCodePointsSize;
+      return;
+    }
+    std::vector<int32_t> distances;
+    distances.reserve(rightCodePointsSize);
+    // These fills ensure that the value above the rightmost entry of our
+    // stripe will be ignored in following loop iterations.
+    int32_t boundary = std::min<int32_t>(rightCodePointsSize, threshold);
+    auto i = 0;
+    for (; i < boundary; i++) {
+      distances.push_back(i + 1);
+    }
+    for (; i < rightCodePointsSize; i++) {
+      distances.push_back(INT32_MAX);
+    }
+
+    for (auto i = 0; i < leftCodePointsSize; i++) {
+      auto lower = std::max<int32_t>(0, i - threshold);
+      int32_t maxValueWithThreshold;
+      int32_t upper = rightCodePointsSize;
+      if (!__builtin_add_overflow(i + 1, threshold, &maxValueWithThreshold)) {
+        upper = std::min<int32_t>(rightCodePointsSize, maxValueWithThreshold);
+      }
+      if (lower > upper) {
+        result = -1;
+        return;
+      }
+      int32_t leftUpDistance;
+      if (lower == 0) {
+        leftUpDistance = distances[lower];
+        if (leftCodePoints[i] == rightCodePoints[0]) {
+          distances[0] = i;
+        } else {
+          distances[0] = std::min(i, distances[0]) + 1;
+        }
+        lower = 1;
+      } else {
+        leftUpDistance = distances[lower - 1];
+        // Ignore entry left of leftmost.
+        distances[lower - 1] = INT32_MAX;
+      }
+      for (int j = lower; j < upper; j++) {
+        auto leftUpDistanceNext = distances[j];
+        if (leftCodePoints[i] == rightCodePoints[j]) {
+          distances[j] = leftUpDistance;
+        } else {
+          distances[j] =
+              std::min(
+                  distances[j - 1], std::min(leftUpDistance, distances[j])) +
+              1;
+        }
+        leftUpDistance = leftUpDistanceNext;
+      }
+    }
+    result = distances[rightCodePointsSize - 1];
+    if (result > threshold) {
+      result = -1;
```

Comment:
return?

Reply:
Here we already reach the end of function, I think we don't need to add explicit `return` statement. Please let me know if things I missed here.